docs/writing-your-first-rclrs-node.md

```diff
@@ -0,0 +1,214 @@
+
+> 💡 As an aside, this requirement is expressed by the `'static` bound on the generic parameter `F` for the callback in `Node::create_subscription()`.
+
+You might think "I don't want to borrow from the local variable `data` anyway, I want to borrow from the `data` field in `RepublisherNode`!" and you would be right. That variable lives considerably longer, but also not forever, so it wouldn't be enough. A secondary problem is that it would be a *self-referential struct*, which is not allowed in Rust.
+
+The solution is _shared ownership_ of the data by the callback and the node. The `Arc` type provides shared ownership, but since it only gives out shared references to its data, we also need a `Mutex` or a `RefCell`. This `Arc<Mutex<T>>` type is a frequent pattern in Rust code.
+
+So, to store the received data in the struct, the following things have to change:
+1. Import `Mutex`
+2. Adjust the type of the `data` field
+3. Create two pointers to the same data (wrapped in a `Mutex`)
+4. Make the closure `move`, and inside it, lock the `Mutex` and store the message  
+
+```rust
+use std::sync::{Arc, Mutex};  // (1)
+use std_msgs::msg::String as StringMsg;
+
+struct RepublisherNode {
+    node: rclrs::Node,
+    _subscription: Arc<rclrs::Subscription<StringMsg>>,
+    data: Arc<Mutex<Option<StringMsg>>>,  // (2)
+}
+
+impl RepublisherNode {
+    fn new(context: &rclrs::Context) -> Result<Self, rclrs::RclrsError> {
+        let mut node = rclrs::Node::new("republisher", context)?;
+        let data = Arc::new(Mutex::new(None));  // (3)
+        let data_cb = Arc::clone(&data);
+        let _subscription = {
+            // Create a new shared pointer instance that will be owned by the closure
+            node.create_subscription(
+                "in_topic",
+                rclrs::QOS_PROFILE_DEFAULT,
+                move |msg: StringMsg| {
+                    // This subscription now owns the data_cb variable
+                    *data_cb.lock().unwrap() = Some(msg);  // (4)
+                },
+            )?
+        };
+        Ok(Self {
+            node,
+            _subscription,
+            data,
+        })
+    }
+}
+```
+
+If that seems needlessly complicated – maybe it is, in the sense that `rclrs` could potentially introduce new abstractions to improve the ergonomics of this use case. This is to be discussed.
+
+If you couldn't follow the explanation involving borrowing, closures etc. above, an explanation of these concepts is unfortunately out of scope of this tutorial. There are many good Rust books and tutorials that can help you understand these crucial features.
+
+## Periodically run a republishing function
+
+The node still doesn't republish the received messages. First, let's add a publisher to the node:
+
+```rust
+// Add this new field to the RepublisherNode struct:
+publisher: rclrs::Publisher<StringMsg>,
```

Comment:
Tiny detail, I noticed based on how you construct the struct later that this needs to be placed between the _subscription and data fields. Maybe update this to show the whole new struct with this line added to make it clearer?